ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -371,12 +376,64 @@ def add_capacity(osd_size_capacity_requested):
     storageDeviceSets->count = (capacity reqested / osd capacity ) + existing count storageDeviceSets
 
     """
-
+    osd_size_existing = get_osd_size()
+    storage_devices_sets_required = int(osd_size_capacity_requested / osd_size_existing)
+    lvpresent = check_local_volume()
+    if lvpresent:
+        ocp_obj = OCP()
+        workers = get_typed_nodes(node_type='worker')
+        worker_names = [worker.name for worker in workers]
+        output = ocp_obj.exec_oc_cmd("get localvolume local-block -n local-storage -o yaml")
+        cur_device_list = output["spec"]["storageClassDevices"][0]['devicePaths']
+        path = os.path.join(constants.EXTERNAL_DIR, "device-by-id-ocp")
+        utils.clone_repo(constants.OCP_QE_DEVICEPATH_REPO, path)
+        os.chdir(path)
+        utils.run_cmd("ansible-playbook devices_by_id.yml")
+        f = open("local-storage-block.yaml", "r")
+        f2 = open("local-block.yaml", "w")
+        w1, w2, w3 = storage_devices_sets_required
+        cur_line = f.readline()
+        while "devicePaths:" not in cur_line:
+            f2.write(cur_line)
+            cur_line = f.readline()
+        f2.write(cur_line)
+        cur_line = f.readline()
+        while cur_line:
+            if str(osd_size_capacity_requested) in cur_line:
+                if w1 and (str(worker_names[0]) in cur_line):
+                    if not any(str(s) in str(cur_line) for s in cur_device_list):
+                        f2.write(cur_line)
+                        w1 = w1 - 1
+                if w2 and (str(worker_names[1]) in cur_line):
+                    if not any(str(s) in str(cur_line) for s in cur_device_list):
+                        f2.write(cur_line)
+                        w2 = w2 - 1
+                if w3 and (str(worker_names[2]) in cur_line):
+                    if not any(str(s) in str(cur_line) for s in cur_device_list):
+                        f2.write(cur_line)
+                        w3 = w3 - 1
+            cur_line = f.readline()
+        f.close()
+        f2.close()
+        obj = open("local-block.yaml")
+        lvcr = yaml.load(obj, Loader=yaml.FullLoader)
```

Comment:
there should be assert generated when enough disk is not available

Reply:
Added